stock_request_analytic/models/stock_request.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2017-2020 ForgeFlow, S.L. (https://www.forgeflow.com)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class StockRequest(models.Model):
+    _inherit = "stock.request"
+
+    analytic_account_id = fields.Many2one(
+        "account.analytic.account", string="Analytic Account"
+    )
+
+    @api.constrains("analytic_account_id")
+    def _check_analytic_company_constrains(self):
+        if any(
+            r.company_id
+            and r.analytic_account_id
+            and r.analytic_account_id.company_id != r.company_id
+            for r in self
+        ):
+            raise ValidationError(
+                _(
+                    "You cannot link a analytic account to a stock request "
+                    "that belongs to another company."
+                )
+            )
```

Comment:
Could this be replaced by adding `check_company=True` to the `analytic_account_id` field?
```suggestion
class StockRequest(models.Model):
    _inherit = "stock.request"
    _check_company_auto = True

    analytic_account_id = fields.Many2one(
        "account.analytic.account", string="Analytic Account", check_company=True
    )
```